### DISCUSSION TITLE
Turning simple JSON-LD into mangle facts?

### DISCUSSION THREAD

**Discussion by burakemir** [google/mangle#10] [General]
People claim that the meaning of JSON-LD is RDF. 

Let's look at this friend-of-a-friend example.
```
{
  "@context": {
    "name": "http://xmlns.com/foaf/0.1/name",
    "homepage": {
      "@id": "http://xmlns.com/foaf/0.1/workplaceHomepage",
      "@type": "@id"
    },
    "Person": "http://xmlns.com/foaf/0.1/Person"
  },
  "@id": "https://me.example.com",
  "@type": "Person",
  "name": "John Smith",
  "homepage": "https://www.example.com/"
}
```

I'd rather do someting like this, and in Mangle I can:

```
foaf_person(/me.example.com).
foaf_name(/me.example.com, "John Smith").
foaf_workplace_homepage(/me.example.com, "https://www.example.com/").
```

It is clear that somewhere, we'd have to remember that predicate `foaf_name` is really "http://xmlns.com/foaf/0.1/name" etc.

I am wondering if one could come up with a full mapping (turning lists to lists, and using multiple facts for sets).

**Comment by vorburger** [comment]
Just FYI to anyone ever seeing this, I intend to explore doing (something like) this in https://github.com/enola-dev/enola/issues/842...
